### ISSUE TITLE
4K output via USB C (thunderbolt port)

### ISSUE THREAD

**Issue by sathishkumar294** [Lovely-XPP/Dell-Latitude-E7480-Hackintosh#11]
Hi @Lovely-XPP, thanks a lot for your great work in getting everything running for Dell 7480 laptop. I had a working EFI, but yours is  just better. Only one shortcoming I notice is that when connected to a 4K monitor via thunderbolt port, the hackintosh recognises only 2560x1600 resolution instead of full 3840x2160. 
I am trying various guides available online, but couldn't get it working so far. If you know how to solve this issue, please do share with me.

Thanks once again!

**Comment by sathishkumar294** [comment]
![Screenshot 2021-12-18 at 1 09 17 AM](https://user-images.githubusercontent.com/6660993/146599087-e333faa7-8a4e-474b-8157-a89ba254ae20.png)
You can see that the external monitor (HP Envy 27" 4K) does not show full resolution. The port works fine when using Windows or Ubuntu

**Comment by Lovely-XPP** [comment]
Ok, you may try the app here.

[RDM.app.zip](https://github.com/Lovely-XPP/Dell-Latitude-E7480-Hackintosh/files/7739077/RDM.app.zip)

**Comment by Lovely-XPP** [comment]
Open the app and there will be a icon in your menu bar like this. Look at it and check if there is 3840*2160 resolution.


<img width="277" alt="image" src="https://user-images.githubusercontent.com/66028151/146637294-39869bf0-9dae-4e43-824e-8186f7718a9a.png">
.

**Comment by sathishkumar294** [comment]
Thanks for your reply @Lovely-XPP. I tried using  the RDM app, but the max resolution shown for my external display is still 2560x1600 instead of 3840x2160. Using ubuntu or windows 10, the thunderbolt port provides full 4K resolution.

**Comment by Lovely-XPP** [comment]
Okay, Maybe your laptop haven't enabled the 3840x2160 resolution. You can try the [one-key-hidpi](https://github.com/xzhih/one-key-hidpi). And enable the 3840x2160 resolution through it.

**Comment by aancw** [comment]
I'm using 4K monitor too but stuck at 2560.

**Comment by Lovely-XPP** [comment]
I don't have a 4k monitor, so I can't know the problem exactly. If I have chance to own one, I will have a try and try to solve the issue.

**Comment by sathishkumar294** [comment]
I tried various other solutions recommended on web, none seem to work. For me it seems more like a MacOS Big Sur problem than something to do with hackintosh. Too bad, I do not have a real mac to confirm this.

**Comment by sathishkumar294** [comment]
I have upgraded to latest Monterey release and the issue still persists.

**Comment by Lovely-XPP** [comment]
Another possibility is that there is a problem with the Thunderbolt's driver in hackintosh.

**Comment by sathishkumar294** [comment]
Solved!. I asked the same question at [osxlatitude](https://osxlatitude.com/forums/topic/17429-latitude-7480-unable-to-obtain-4k-resolution-on-external-monitor-through-tb-dp-port/#comment-115398) and got the fix from an admin. I had to set the DVMT to 64MB through mod grub shell. See [here](https://osxlatitude.com/forums/topic/13963-changing-dvmt-values-in-bios/) for instructions on how to find the variable and set it to 64MB. After that I had to remove the existing Framebuffer patches of stolenmem and fbmem from config.plist. After a reboot, I have now got 4K@60Hz on my 4K monitor.
